subxt/src/client/light_client/builder.rs

```diff
@@ -251,10 +251,17 @@ async fn fetch_url(url: impl AsRef<str>) -> Result<serde_json::Value, Error> {
     use jsonrpsee::core::client::ClientT;
     let client = jsonrpsee_helpers::client(url.as_ref()).await?;
 
-    client
+    let result = client
         .request("sync_state_genSyncSpec", jsonrpsee::rpc_params![true])
         .await
-        .map_err(|err| Error::Rpc(crate::error::RpcError::ClientError(Box::new(err))))
+        .map_err(|err| Error::Rpc(crate::error::RpcError::ClientError(Box::new(err))));
+
+    // Sleep 8 seconds to ensure that a new block is produced for the substrate test node.
+    // Although the block production should be 6 seconds, the CI environment is sometimes slow.
+    // This is a temporary workaround for: https://github.com/smol-dot/smoldot/issues/1562.
```

Comment:
This is good for now.

One thing to keep in mind is that parachains have eg a 12 second block time atm, so in the general case we should probably do something like download chain spec, and then subscribe to finalized blocks and wait for the next one or something like that, to make this nice and robust to block times or slow CI or whatever!

Reply:
(it may also be that if we do this, we don't need tokio/time any more?)